File: module_3/src/main/java/ua/com/alevel/persistence/type/BankOperationType.java
package ua.com.alevel.persistence.type;

public enum BankOperationType {

    INCOME_SALARY("salary", OperationType.INCOME),
    INCOME_OTHER("other income", OperationType.INCOME),
    INCOME_TRANSFER("transfer to card", OperationType.INCOME),

    OUTCOME_PAYMENT("payment", OperationType.EXPENSE),
    OUTCOME_TRANSFER("transfer from card", OperationType.EXPENSE),
    OUTCOME_CREDIT("loan repayment", OperationType.EXPENSE),
    OUTCOME_OTHER("other expense", OperationType.EXPENSE);

    private final String name;
    private final OperationType type;

    BankOperationType(String name, OperationType type) {
        this.name = name;
        this.type = type;
    }
}

File: module_3/module-client/src/app/model/bank-operation-type.ts

export enum BankOperationType {
  INCOME_SALARY = 'INCOME_SALARY',
  INCOME_OTHER = 'INCOME_OTHER',
  INCOME_TRANSFER = 'INCOME_TRANSFER',

  OUTCOME_PAYMENT = 'OUTCOME_PAYMENT',
  OUTCOME_TRANSFER = 'OUTCOME_TRANSFER',
  OUTCOME_CREDIT = 'OUTCOME_CREDIT',
  OUTCOME_OTHER = 'OUTCOME_OTHER'

}

export const BankOperationTypeMapping = {
  [BankOperationType.INCOME_SALARY]: 'salary',
  [BankOperationType.INCOME_OTHER]: 'other income',
  [BankOperationType.INCOME_TRANSFER]: 'transfer to card',
  [BankOperationType.OUTCOME_PAYMENT]: 'payment',
  [BankOperationType.OUTCOME_TRANSFER]: 'transfer from card',
  [BankOperationType.OUTCOME_CREDIT]: 'loan repayment',
  [BankOperationType.OUTCOME_OTHER]: 'other expense'
};

export const IncomeBankOperationTypeMapping = {
  [BankOperationType.INCOME_SALARY]: 'salary',
  [BankOperationType.INCOME_OTHER]: 'other income'
};

export const OutcomeBankOperationTypeMapping = {

  [BankOperationType.OUTCOME_PAYMENT]: 'payment',
  [BankOperationType.OUTCOME_TRANSFER]: 'transfer from card',
  [BankOperationType.OUTCOME_CREDIT]: 'loan repayment',
  [BankOperationType.OUTCOME_OTHER]: 'other expense'
};
